src/main/java/org/manifold/compiler/front/FunctionInvocationVertex.java

```diff
@@ -0,0 +1,117 @@
+package org.manifold.compiler.front;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.manifold.compiler.NodeTypeValue;
+import org.manifold.compiler.TypeValue;
+import org.manifold.compiler.UndefinedBehaviourError;
+import org.manifold.compiler.Value;
+
+public class FunctionInvocationVertex extends ExpressionVertex {
+
+  private static Logger log = LogManager.getLogger("FunctionInvocationVertex");
+
+  private ExpressionEdge functionEdge;
+  public ExpressionEdge getFunctionEdge() {
+    return functionEdge;
+  }
+
+  private ExpressionEdge inputEdge;
+  public ExpressionEdge getInputEdge() {
+    return inputEdge;
+  }
+
+  public FunctionInvocationVertex(ExpressionGraph exprGraph,
+      ExpressionEdge functionEdge, ExpressionEdge inputEdge) {
+    super(exprGraph);
+    this.functionEdge = functionEdge;
+    this.functionEdge.setTarget(this);
+    this.functionEdge.setName("function");
+    this.inputEdge = inputEdge;
+    this.inputEdge.setTarget(this);
+    this.inputEdge.setName("argument");
+  }
+
+  @Override
+  public TypeValue getType() {
+    // TODO Auto-generated method stub
+    return null;
```

Comment:
I'm still trying to decide whether I think these methods are still necessary to have around